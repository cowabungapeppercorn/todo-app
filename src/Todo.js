import React from 'react';
import EditTodoForm from './EditTodoForm';

class Todo extends React.Component {
  static defaultProps = {
    todoInfo: {
      text: "you should type things",
      id: "this is not that unique",
      editMode: false
    }
  }

  render() {

    let { text, id, editMode } = this.props.todoInfo;

    if (editMode) {
      return (
        <li>
          <EditTodoForm editTodo={this.props.editTodo} id={id} text={text} />
          editing...
          <button onClick={() => this.props.remove(id)}>X</button>
        </li>
      );
    } else {
      return (
        <li>
          {text}
          <button onClick={() => this.props.remove(id)}>X</button>
          <button onClick={() => this.props.toggleEdit(id)}>Edit</button>
        </li>
      );
    }
  }
}

export default Todo;