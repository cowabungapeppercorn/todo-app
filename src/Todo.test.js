import React from 'react';
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json';
import Todo from './Todo';

it('renders without crashing', () => {
  shallow(<Todo />);
});

it('renders as expected', () => {
  let todoObj = {
    text: "yo",
    id: 5,
    editMode: false
  };
  let wrapper = mount(<Todo todoInfo={todoObj} />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
})