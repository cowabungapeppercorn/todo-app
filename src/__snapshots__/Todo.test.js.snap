// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<Todo
  todoInfo={
    Object {
      "editMode": false,
      "id": 5,
      "text": "yo",
    }
  }
>
  <li>
    yo
    <button
      onClick={[Function]}
    >
      X
    </button>
    <button
      onClick={[Function]}
    >
      Edit
    </button>
  </li>
</Todo>
`;
